src/main/scala/nl.knaw.dans.easy.bagstore/component/FileSystemComponent.scala

```diff
@@ -120,14 +115,17 @@ trait FileSystemComponent extends DebugEnhancedLogging {
           Failure(IncompleteItemUriException("base-uri by itself is not an item-uri"))
         else {
           val uuidStr = formatUuidStrCanonically(itemIdPath.getName(0).toString.filterNot(_ == '-'))
-          assertUuidValid(uuidStr)
-          val bagId = BagId(UUID.fromString(uuidStr))
-          if (itemIdPath.getNameCount > 1)
-            Try(FileId(bagId, itemIdPath.subpath(1, itemIdPath.getNameCount)))
-          else if (uri.toString.endsWith("/"))
-            Try(FileId(bagId, Paths.get("")))
-          else
-            Try(bagId)
+          uuidStr.toUUID.toTry match {
+            case Success(uuid) =>
+              val bagId = BagId(uuid)
+              if (itemIdPath.getNameCount > 1)
+                Try(FileId(bagId, itemIdPath.subpath(1, itemIdPath.getNameCount)))
+              else if (uri.toString.endsWith("/"))
+                     Try(FileId(bagId, Paths.get("")))
+              else
+                Try(bagId)
+            case Failure(e) => Failure(new IllegalArgumentException(e.getMessage))
+          }
```

Comment:
nice rewrite! This opens the door to more improvements: For example, you don't need the `Try()` wrappers around `FileId` if you use `.map(...).recoverWith { ... }` instead of the pattern matches. These wrappers were there not to catch errors, but to satisfy the types...

```scala
uuidStr.toUUID.toTry
  .map {
    case uuid if itemIdPath.getNameCount > 1 => FileId(BagId(uuid), itemIdPath.subpath(1, itemIdPath.getNameCount))
    case uuid if uri.toString.endsWith("/") => FileId(BagId(uuid), Paths.get(""))
    case uuid => BagId(uuid)
  }
  .recoverWith { case e => Failure(new IllegalArgumentException(e.getMessage, e))
```

Reply:
beautiful